src/components/Compare/CompareMain.tsx

```diff
@@ -146,72 +145,41 @@ const CompareMain = (props: {
     countyTypeToView,
     viewAllCounties,
     geoScope,
+    stateId,
+    countyId: county?.full_fips_code,
```

Comment:
I used to handle stateId / countyId specially below since they could be undefined, but instead I reworked the storeSharedComponentParams() function to deal with undefined better (Firestore doesn't support storing `undefined` explicitly).